### DISCUSSION TITLE
How to set background color for mini player

### DISCUSSION THREAD

**Discussion by fury21** [LeoNatan/LNPopupController#569] [General] [answered]
Hello. I want to set the background color of the mini player to red for example, here's how I do it:

 private func setupPlayerPopUpBar() {
        let appearance = LNPopupBarAppearance()
        let titleTextAttributesForegroundColor = UIColor.Scheme.Label.text8()
        let subtitleTextAttributesForegroundColor = UIColor.Scheme.Label.text9()
        
        appearance.__titleTextAttributes = [.font: UIFont(name: "Jost-SemiBold", size: 14)!,
                                            .foregroundColor: titleTextAttributesForegroundColor]
        appearance.__subtitleTextAttributes = [.font: UIFont(name: "Jost-Regular", size: 12)!,
                                               .foregroundColor: subtitleTextAttributesForegroundColor]
        
//        appearance.backgroundEffect = UIBlurEffect(style: .dark)
----- here --->        appearance.backgroundColor = .red 

        let appearanceProxy = LNPopupBar.appearance(whenContainedInInstancesOf: [UINavigationController.self])
        appearanceProxy.inheritsAppearanceFromDockingView = false
        appearanceProxy.standardAppearance = appearance
        appearanceProxy.tintColor = .pink1_FF0B77()
        appearanceProxy.progressViewStyle = .top

        popupBar.barStyle = .compact
        
        popupInteractionStyle = .drag
        popupBar.imageView.contentMode = .scaleAspectFill
    }

...but I get a red background that goes beyond the mini-player and its height is half the height of the mini-player. How can I set a non-transparent red background for the minibar?

Also, how can I change the background color of the minibar from red to green when the minibar is visible on the screen?

<img width="446" alt="Снимок экрана 2024-05-13 в 20 46 14" src="https://github.com/LeoNatan/LNPopupController/assets/30409500/ae919c43-c3d5-4df5-a249-3557a6764b53">

**Comment by LeoNatan** [comment, marked as answer]
You are using the floating bar style. Set the `floatingBackgroundColor` property instead.

**Reply by fury21** [reply]
This helped, thanks! 

How can I change the background if the mini-player is visible on the screen? Now I change appearance.floatingBackgroundColor, but the changes are only visible if I close the mini-player and open it again?

**Reply by LeoNatan** [reply]
Like I said in the other discussion, do not use the legacy appearance proxy API. Set the properties directly on the appropriate bar.

**Reply by LeoNatan** [reply]
I suggest you look in the example app for how to use the API.

**Comment by divyeshdobariya** [comment]
Can we change floatingBackgroundColor once already set while open popup, I want to change on Button Click from other ViewController.

**Reply by LeoNatan** [reply]
Yes, it should be possible.

**Reply by divyeshdobariya** [reply]
I tried so many solutions, but did not work , here is the below solutions i tried

popupBar.standardAppearance.floatingBackgroundColor = .red

is there anything still remain to add ? please help , I am trying different way from last 10~12 hours, but did not work anything.

Here is my code that i write to setup appearance : https://github.com/LeoNatan/LNPopupController/discussions/587

**Reply by LeoNatan** [reply]
Make sure you are accessing the correct popup bar instance. You should access the popup **container** controller’s popup bar. 

For ease, the popup item exposes appearance controls too. Perhaps it would be easier for you to modify those values.

**Reply by divyeshdobariya** [reply]
Can you give me that code for that, because i didn't understand how to setup that code.. Your help will be blessed for me.. Thanks

**Reply by LeoNatan** [reply]
Sorry, I am on vacation and do not have access to code. I’ll be back on March 5th.

**Reply by divyeshdobariya** [reply]
Thanks for your reply , i will try until you back, Once you will be back give me suggestion and code, Enjoy your vacation :)

**Reply by LeoNatan** [reply]
Is this what you are trying to achieve?

![colors](https://github.com/user-attachments/assets/5ea2682d-01e3-44fa-b32a-b3c233b5a361)

Assuming you are trying to make this happen from the content view controller (the view controller inside the popup), this is an example of a way to achieve this:

```swift
self.popupPresentationContainer?.popupBar.standardAppearance.configureWithOpaqueFloatingBackground()
self.popupPresentationContainer?.popupBar.standardAppearance.floatingBackgroundColor = LNRandomSystemColor();
```

**Reply by divyeshdobariya** [reply]
Thanks for your great support, Now it's working fine :) , you are the best !!! .... Thankssss

**Reply by LeoNatan** [reply]
Happy coding
